test/polytopes.jl

```diff
@@ -302,6 +302,10 @@ end
   @test vertices(Ngon{3}(pts...)) == verts
   @test vertices(Ngon{3}(tups...)) == verts
 
+  @test collect(eachvertex(Ngon(pts))) == verts
+  @test @allocated eachvertex(Ngon(pts)) == 0
+  @test @allocated Base.iterate(eachvertex(Ngon(pts))) == 0
```

Comment:
Can we replace this test by a test with a for loop?

```julia
@test @allocated begin
  for v in eachvertex(ngon)
    # do nothing
  end
end 0
```

Reply:
I've quickly checked with grids and multi, but those still allocate, so I didn't put it for those since setting a magic number of allocation felt finnicky.
but for polytopes it should be always non allocating so it should be fine.